go-app-ussd.js

```diff
@@ -550,6 +550,127 @@ go.utils = {
             });
     },
 
+    nightingale_categories: function(im, contact) {
+        return [
+            im.config.nightingale.category[contact.extra.report_theme],
+            im.config.nightingale.category[contact.extra.report_category]
+        ];
+    },
+
+    nightingale_api_call: function (method, params, payload, endpoint, im) {
+        var http = new JsonApi(im, {
+            headers: {
+                'Authorization': ['Token ' + im.config.nightingale.api_key]
+            }
+        });
+        switch (method) {
+            case "post":
+                return http.post(im.config.nightingale.api_root + endpoint, {
+                    data: payload
+                });
+            case "get":
+                return http.get(im.config.nightingale.api_root + endpoint, {
+                    params: params
+                });
+            case "patch":
+                return http.patch(im.config.nightingale.api_root + endpoint, {
+                    data: payload
+                });
+            case "put":
+                return http.put(im.config.nightingale.api_root + endpoint, {
+                    params: params,
+                  data: payload
+                });
+            case "delete":
+                return http.delete(im.config.nightingale.api_root + endpoint);
+            }
+    },
+
+    nightingale_get_report: function(im, contact) {
+        var method = "get";
+        var params = null;
+        var endpoint = "report/" + contact.extra.last_report_id + "/";
+        var payload = null;
+        return go.utils
+            .nightingale_api_call(method, params, payload, endpoint, im)
+            .then(function(response) {
+                return response;
+            });
+    },
+
+    nightingale_patch: function(im, contact) {
+        var method = "patch";
+        var params = null;
+        var endpoint = "report/" + contact.extra.last_report_id + "/";
+        return go.utils
+            .nightingale_get_report(im, contact)
```

Comment:
You don't need to get the report to patch it, just patch with `{"description": "Blah"}`